### DISCUSSION TITLE
Site pessoal para desenvolvedor?

### DISCUSSION THREAD

**Discussion by wellingtonpgp** [frontendbr/forum#440] [Discussão]
Galera preciso criar um site pessoal, o que seria legal colocar nele? Qual estilo devo seguir minimalista ou "cheio de perfumarias"? Valeu Xovens! 🚀

**Comment by RatoX** [comment]
Cara, eu venho desenvolvendo o meu site, a passos lentos, só com o basico de firula no css porém com tudo o que tem de "novo" webpack + mocha + ES6 + babel + inline css + 60 fps tudo em VanilaJS mesmo.

Vou entrar de férias semana que vem e vou me focar nele pra colocar mais firulas de frontend:

- Trocar o mocha pelo AVA;
- Colocar o nightwatch;
- Colocar mais informações sobre minha experiência profissional e essas coisas;

Saca só o resultado temporario: https://ratox.github.io/about-me/
E o repo tem umas coisas legais lá: https://github.com/RatoX/about-me

**Comment by fernandosouza** [comment]
Estou em débito comigo mesmo em relação a isso. Mas meu próximo site pessoa pretendo deixa-lo extremamente clean, reportando apenas os projetos que trabalhei com links para o código no Github.

**Comment by matheusml** [comment]
Cara, pode usar o meu como referência:
https://matheuslima.com/

**Comment by brunoravanhani** [comment]
Cara, site pessoal é tenso, fica sempre pra depois... masse vai fazer acho que depende muito do que quer com ele, se é um blog, um portifólio, curriculo, etc., mas é sempre legal impressionar né tipo esse [curriculo aqui](http://torto.github.io/curriculo/) do @torto, mas se teu foco for mais sério ai menos é mais.

**Comment by andresantos-tech** [comment]
Eu por exemplo tenho um site pessoal em planejamento há quase 10 anos =)

Acredito que nesse caso "fazer" é mais importante do que "como fazer". Exigência alta mais tecnologia que evolui diariamente em geral resulta em deixar sempre pra depois. Muito provavelmente você vai mexer nele pra sempre, então não tem porque perder muito tempo na versão inicial.

Faz o layout mais clean possível, coloca as informações que têm relevância pra sua carreira, e se for escrever sobre tecnologia (ou qualquer outro assunto) integre com o Medium. O site do @matheusml é uma ótima referência nesse estilo.

**Comment by RatoX** [comment]
O site do @torto é o melhor, AHAHAHHA.

Só falta ele ajeitar pra ficar legal no mobile!

**Comment by giioohbernini** [comment]
Me bateu uma duvida recentemente muito relacionada a essa issue, então vou aproveitar o embalo! 

Tem pessoas que publicam em seu próprio site/blog, e outras que preferem postar no medium e linkar no seu site, como exemplo, o já citado @matheusml 

O que vocês acham disso? Eu particularmente gosto bastante do medium, mas queria a opinião de vocês sobre essa prática.

**Comment by RatoX** [comment]
@giioohbernini Eu curto meu próprio site pq eu posso colocar todas as técnicas de desenvolvimento em prática.

**Comment by hugocarreira** [comment]
@giioohbernini eu, particularmente, gosto muito de fazer meu próprio site/blog, pois é uma maneira de, assim com o @RatoX disse, colocar as técnicas em prática.

Quando eu estava estudando MaterializeCSS, fiz meu [site](http://site.hugocarreira.me) e meu [blog](http://hugocarreira.me) usando ele. O site foi apenas uma brincadeira, um teste.. fiz em uma noite e nunca mais mexi. O blog eu curto muito ele até hoje, disponibilizei o [tema](https://github.com/hugocarreira/jekyll-materializecss) e ganhei alguns adeptos. 

Resumindo, fazer o site é legal...
E, diferente de mim, ir atualizando-o com tecnologia novas é a parte mais interessante.

**Comment by thulioph** [comment]
Eu indicaria você por em prática sua idéia e depois que ela estiver pronta você vai ajeitando, etc.. O importante é não ficar parado, postergando.

Sobre postar em outros canais fora do seu site, eu não curto muito, um dia aquela plataforma pode ficar off ou acabar e meu conteúdo vai junto.. Se puder, escreva no seu blog e replique no Medium (por questões de alcance).

Se quer começar a gerar conteúdo rápido e sem burocracia, utilize algum gerador estático como por exemplo o [Hexo](https://hexo.io/), além de ter uma porrada de temas você pode criar o seu, vale a pena conferir.

Como muitos falaram, depende do que você quer, comece devagar, mas seja constante.

**Comment by rubenmarcus** [comment]
@RatoX  Inline-css é tendencia nova?

**Comment by rubenmarcus** [comment]
Eu ia abrir um tópico parecido sobre portfolios, no meu ver você tem que ver outros de referencia, esse é o meu fiz em 2014 em Angular, está cheio de gambiarras e não está finalizado,

http://rubenmarcus.com/


Vou fazer uma nova versão em Vue, mas estava pensando em deixa-lo em algum cms, alguem recomemda algum cms pra isso? Pensei até em montar em algo em Node ( nunca mexi ) , ou algum em PHP ( wordpress, grav)



gosto muito do portfolio desse italiano: 
http://claudiocalautti.cc/


Eu piro em ver animações,jogos,templates e etc, pq vim do webdesign, mas tem gente que não tem um portfolio clean e animado, e sim um template de bootstrap ou as vezes só os repositorios do git, tudo depende a vaga que voce concorre, seu nivel etc, pra UX ,vi que a maioria dá mais importancia pra um Behance do que pra um Git..

agora

pra front no meu ver, pelo menos esse ano eu quero:

- Ter um CodePen cheio de códigos teste e animações
- um git com repositorios do que desenvolvi
- um site e blog/proprio, assim como um medium
- templates wordpress, alguns free e alguns pro
- exemplos e tutoriais de js e frameworks js.. 
- alguns cases de projetos que fiz..


tem que ter tempo.... e dedicação.. mas acho que o começo é buscar as referencias.. aqui no forum mesmo tem várias @willianjusten  , @LFeh  ...

**Comment by RatoX** [comment]
@rubenmarcus ahhahaha pra perfomance é cara.

**Comment by raulfernando** [comment]
Mas não é tendencia nova

**Comment by felipefialho** [comment]
Mano, depende muito do seu estilo.

Sou minimalista na vida, e não podia deixar de levar isso para meu site. O principal é o conteúdo, e não o layout, que pra mim é só um facilitador de acesso ao que importa. Quanto melhor é o layout, mais simples torna-se a navegação, usabilidade... e colocar perfumarias tende a jogar contra.

Foquei principalmente em performance (tudo estático, JavaScript SÓ no GA e no Prism).

http://www.felipefialho.com/

**Comment by raulfernando** [comment]
Então acho que esta sendo uma tendencia agora a galera criar uns layout minimalista, muitos dev's os blog estão neste estilo e eu particularmente acho bem legal

Como o @LFeh  disse o importante é o conteúdo. Acho que esta é a receita postar um conteúdo bacana e com frequência e se fizer um layout minimalista melhor ainda

**Comment by rubenmarcus** [comment]
no que o css inline ajuda em perfomance? usar cache nao seria melhor?? existe algum tutorial ou artigo falando sobre isso? Tudo que li até hoje vai contra essa tecnica, ai queria ler melhor pra entender, existe algo que gera isso?

**Comment by felipefialho** [comment]
@rubenmarcus Abre uma issue pra discutir isso :)

**Comment by raulfernando** [comment]
Então @rubenmarcus tem uma galera que coloca o css inicial da pagina inline para não precisar esperar baixar a folha de estilo e assim exibir mais rápido o topo do site. Massssss nunca vi ninguém fazer isso só ouvi boates de que é mais rápido colocar o css do topo inline 

neste [post](http://blog.caelum.com.br/performance-web-no-mundo-real-porque-o-site-do-alura-voa/) do sergio lopes ele fala 

> O consenso do povo de performance Web hoje em dia é que o foco deve ser remover todo e qualquer recurso blocante do carregamento inicial da página. É o que o pessoal chama de otimizar o Critical Path. E o maior inimigo aqui é o CSS. Ele fica no <head> e bloqueia a renderização. Chegam até a recomendar não usar um arquivo .css e embutir o código CSS inicial numa tag <style> inline. Tudo pra destravar o critical path.

Acho que é isso me corrijam se eu estiver errado

**Comment by kazzkiq** [comment]
@raulfernando É isso mesmo, por padrão o navegador bloqueia a renderização da página até baixar os arquivos CSS referenciados nessa página. Dependendo do tamanho desses arquivos CSS (ou da conexão do usuário) isso pode acabar refletindo em uma renderização demorada (aquela famosa página branca enquanto o ícone de *loading* fica girando infinitamente até tirar a paciência do usuário).

Pra prevenir esse tipo de problema, o ideal é que uma página HTML venha com o mínimo CSS necessário **já inbutido no próprio HTML**, permitindo assim que haja interação com o usuário o mais rápido possível, e deixando o restante do CSS para ser carregado mais tarde. Esse pedaço de CSS que viria junto com o HTML é chamado de [Critical CSS](https://www.smashingmagazine.com/2015/08/understanding-critical-css/).

**Comment by raulfernando** [comment]
@kazzkiq explicação perfeita!!! Então mas creio que colocar o css inline so é viável quando o seu arquivo css for muito grande não? Pois nunca vi ninguém fazer isso, quem souber algum site que faz isso seria uma boa compartilhar ai pra vermos

**Comment by RatoX** [comment]
@kazzkiq @raulfernando Aqui tem uma idea bem legal sobre [perfomance.](https://developers.google.com/web/fundamentals/performance/critical-rendering-path/analyzing-crp).

No meu site se [tu olhar eu](view-source:https://ratox.github.io/about-me/) coloco tudo inline já que é pouca coisa ai no primeiro-byte ele já vai de vez tudo, ai depois é só baixar as imagens ( que eu vou colocar em Base64 ou carregar pelo JS já já ). 

Eu consegui deixar o First View abaixo do 1s e o Start Render abaixo do 0.7s minha meta é deixar abaixo do 0.9s. [dados](https://www.webpagetest.org/result/170206_75_16AS/)

<img width="406" alt="screen shot 2017-02-06 at 5 54 37 pm" src="https://cloud.githubusercontent.com/assets/3452187/22663882/97644586-ec95-11e6-87e6-ed72db9b6245.png">

**Comment by rubenmarcus** [comment]
Opa só referenciando a issue #381  
e esse artigo : http://desenvolvimentoparaweb.com/css/css-critico-acima-da-borda-desmistificando/

**Comment by akfzambrana** [comment]
tem este projeto de um cara da nossa comunidade https://github.com/WendellAdriel/dev-curriculum

**Comment by leandroruel** [comment]
o máximo que eu possuo é um codepen com algumas animações que eu fiz. mas tá abandonado a uns meses. preciso voltar a ativa, adorava ficar lá fazendo umas coisas novas

**Comment by kvnol** [comment]
Falando em minimalismo, eu gosto de ser "extreme" às vezes e coloquei minha nova página no ar essa semana. Mandei o blog para o Medium e o portfólio para o Dribbble: [kevinoliveira.com.br](https://kevinoliveira.com.br/).

**Comment by JulianoPadilha** [comment]
@kvnol link do Medium tá dando 404.
